sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CTESubstitution.scala

```diff
@@ -288,13 +288,7 @@ object CTESubstitution extends Rule[LogicalPlan] {
           }
         }.getOrElse(u)
 
-      case p: PlanWithUnresolvedIdentifier =>
-        // We must look up CTE relations first when resolving `UnresolvedRelation`s,
-        // but we can't do it here as `PlanWithUnresolvedIdentifier` is a leaf node
-        // and may produce `UnresolvedRelation` later.
-        // Here we wrap it with `UnresolvedWithCTERelations` so that we can
-        // delay the CTE relations lookup after `PlanWithUnresolvedIdentifier` is resolved.
-        UnresolvedWithCTERelations(p, cteRelations)
+      case p: PlanWithUnresolvedIdentifier => p // Just delay until bindings are done
```

Comment:
do we really need to delay it? It seems no harm to do the CTE relation resolution right now.